src/OrbitGl/GraphTrack.cpp

```diff
@@ -170,3 +227,43 @@ float GraphTrack::GetHeight() const {
                  layout_->GetEventTrackHeight() + layout_->GetTrackBottomMargin();
   return height;
 }
+
+void GraphTrack::SetWarningThresholdWhenEmpty(const std::string& pretty_label, double raw_value) {
+  if (warning_threshold_.has_value()) return;
+  warning_threshold_ = std::make_pair(pretty_label, raw_value);
+  UpdateMinAndMax(raw_value);
+}
+
+void GraphTrack::SetValueUpperBoundWhenEmpty(const std::string& pretty_label, double raw_value) {
+  if (value_upper_bound_.has_value()) return;
+  value_upper_bound_ = std::make_pair(pretty_label, raw_value);
+  UpdateMinAndMax(raw_value);
+}
+void GraphTrack::SetValueLowerBoundWhenEmpty(const std::string& pretty_label, double raw_value) {
+  if (value_lower_bound_.has_value()) return;
+  value_lower_bound_ = std::make_pair(pretty_label, raw_value);
+  UpdateMinAndMax(raw_value);
+}
+
+void GraphTrack::SetLabelUnitWhenEmpty(const std::string& label_unit) {
+  if (!label_unit_.empty()) return;
+  label_unit_ = label_unit;
+}
+
+void GraphTrack::SetValueDecimalDigitsWhenEmpty(uint8_t value_decimal_digits) {
+  if (value_decimal_digits_.has_value()) return;
+  value_decimal_digits_ = value_decimal_digits;
+}
+
+void GraphTrack::UpdateMinAndMax(double value) {
+  max_ = std::max(max_, value);
+  min_ = std::min(min_, value);
+}
+
+std::string GraphTrack::FormatValue(double value) {
+  if (!value_decimal_digits_.has_value()) return std::to_string(value);
+
+  std::stringstream stream;
+  stream << std::fixed << std::setprecision(value_decimal_digits_.value()) << value;
+  return stream.str();
```

Comment:
This can be done with: (Sorry for not realizing this yesterday. I needed to google this myself)
```
absl::StrFormat("%.*f", value_decimal_digits_.value(), value)
```

Which means you can remove the method `FormatValue`. Which is a good thing, because the naming `FormatValue` is a not a good name. It does not tell me anything about what the method does, only some vary vague formatting.

Reply:
Thanks for suggesting a better solution! Changed accordingly!